labconnect/main/routes.py

```diff
@@ -171,20 +363,566 @@ def getLabManagerOpportunityCards() -> dict[Any, list[Any]]:
+            )
+        )
+        data = query.all()
+        print(data)
+
+        if not data or len(data) == 0:
+            abort(404)
+
+        dictionary = data[0][0].to_dict()
+        dictionary["semester"] = dictionary["semester"].upper()
+        dictionary["courses"] = set()
+        dictionary["majors"] = set()
+        dictionary["years"] = set()
+
+        for row in data:
+            dictionary["courses"].add(row[2].course_code)
+            dictionary["majors"].add(row[1].major_code)
+            dictionary["years"].add(row[3].class_year)
+
+        dictionary["courses"] = list(dictionary["courses"])
+        dictionary["majors"] = list(dictionary["majors"])
+        dictionary["years"] = list(dictionary["years"])
+
+        for i in range(len(dictionary["years"])):
+            dictionary["years"][i] = str(dictionary["years"][i])
+
+        dictionary["credits"] = []
+        if dictionary["one_credit"]:
+            dictionary["credits"].append("1")
+
+        if dictionary["two_credits"]:
+            dictionary["credits"].append("2")
+
+        if dictionary["three_credits"]:
+            dictionary["credits"].append("3")
+
+        if dictionary["four_credits"]:
+            dictionary["credits"].append("4")
+
+        dictionary.pop("one_credit")
+        dictionary.pop("two_credits")
+        dictionary.pop("three_credits")
+        dictionary.pop("four_credits")
+
+        return {"data": dictionary}
+
+    abort(500)
+
+
+@main_blueprint.route("/deleteOpportunity", methods=["DELETE", "POST"])
+def deleteOpportunity():
+    if request.method in ["DELETE", "POST"]:
+        data = request.json
+        id = data["id"]
+
+        query = db.session.execute(
+            db.select(
+                Opportunities,
+                RecommendsMajors,
+                RecommendsCourses,
+                RecommendsClassYears,
+                Leads,
+            )
+            .filter(Opportunities.id == id)
+            .join(RecommendsMajors, RecommendsMajors.opportunity_id == Opportunities.id)
+            .join(
+                RecommendsCourses, RecommendsCourses.opportunity_id == Opportunities.id
+            )
+            .join(
+                RecommendsClassYears,
+                RecommendsClassYears.opportunity_id == Opportunities.id,
+            )
+            .join(Leads, Leads.opportunity_id == Opportunities.id)
+        )
+
+        data = query.all()
+        print(data)
+
+        if not data or len(data) == 0:
+            abort(404)
+
+        opportunity = data[0][0]
+
+        for row in data:
+            db.session.delete(row[1])
+            db.session.delete(row[2])
+            db.session.delete(row[3])
+            db.session.delete(row[4])
+
+        leads = data[0][4]
+
+        db.session.delete(opportunity)
+
+        db.session.commit()
+
+        return "Success"
+
+    abort(500)
+
+
+@main_blueprint.route("/changeActiveStatus", methods=["DELETE", "POST"])
```

Comment:
i believe i made an updated version of this in main.opportunity_routes.py already